volmdlr/faces.py

```diff
@@ -5118,455 +5159,381 @@ def is_adjacent(self, face2: 'Face3D'):
 
-    def __init__(self, surface3d: Plane3D, surface2d: Surface2D,
-                 name: str = ''):
-        # if not isinstance(outer_contour2d, volmdlr.Contour2D):
-        #     raise ValueError('Not a contour2D: {}'.format(outer_contour2d))
-        self._bbox = None
-        Face3D.__init__(self,
-                        surface3d=surface3d,
-                        surface2d=surface2d,
-                        name=name)
+            edge_intersections = []
+            for prim1 in self.outer_contour3d.primitives + [prim for inner_contour in self.inner_contours3d
+                                                            for prim in inner_contour.primitives]:
+                edge_intersections = face2.edge_intersections(prim1)
+                if edge_intersections:
+                    return True
+            if not edge_intersections:
+                for prim2 in face2.outer_contour3d.primitives + [prim for inner_contour in face2.inner_contours3d
+                                                                 for prim in inner_contour.primitives]:
+                    edge_intersections = self.edge_intersections(prim2)
+                    if edge_intersections:
+                        return True
 
-    # @classmethod
-    # def _repair_points_and_polygon2d(cls, points, plane):
-    #     if points[0] == points[-1]:
-    #         points = points[:-1]
-    #     polygon_points = [
-    #         p.to_2d(plane.origin, plane.vectors[0], plane.vectors[1]) for p in
-    #         points]
-    #     repaired_points = [p.copy() for p in points]
-    #     polygon2D = volmdlr.ClosedPolygon2D(polygon_points)
-    #     if polygon2D.SelfIntersect()[0]:
-    #         repaired_points = [repaired_points[1]] + [
-    #             repaired_points[0]] + repaired_points[2:]
-    #         polygon_points = [polygon_points[1]] + [
-    #             polygon_points[0]] + polygon_points[2:]
-    #         if polygon_points[0] == polygon_points[-1]:
-    #             repaired_points = repaired_points[:-1]
-    #             polygon_points = polygon_points[:-1]
-    #         polygon2D = volmdlr.ClosedPolygon2D(polygon_points)
-    #     return repaired_points, polygon2D
-
-    # @classmethod
-    # def dict_to_object(cls, dict_, global_dict=None, pointers_memo: Dict[str, Any] = None, path: str = '#'):
-    #     plane3d = Plane3D.dict_to_object(dict_['surface3d'],
-    #                                      global_dict=global_dict,
-    #                                      pointers_memo=pointers_memo,
-    #                                      path=f'{path}/surface3d')
-    #     surface2d = Surface2D.dict_to_object(dict_['surface2d'],
-    #                                          global_dict=global_dict,
-    #                                          pointers_memo=pointers_memo,
-    #                                          path=f'{path}/surface2d')
-    #     return cls(plane3d, surface2d, dict_['name'])
+        return False
```

Comment:
Displaced from Planeface3D to Face3D. Minor changes, but works exactly as before.